packages/features/users/components/UserTable/DeleteMemberModal.tsx

```diff
@@ -13,15 +13,33 @@ export function DeleteMemberModal({ state, dispatch }: { state: State; dispatch:
   const utils = trpc.useUtils();
   const removeMemberMutation = trpc.viewer.teams.removeMember.useMutation({
     onSuccess() {
-      // We don't need to wait for invalidate to finish
-      Promise.all([
-        utils.viewer.teams.get.invalidate(),
-        utils.viewer.eventTypes.invalidate(),
-        utils.viewer.organizations.listMembers.invalidate(),
-      ]);
+      // Update the infinite data to remove the deleted user
+      utils.viewer.organizations.listMembers.setInfiniteData({ limit: 10, searchTerm: "" }, (oldData) => {
+        if (!oldData) return oldData;
+        return {
+          ...oldData,
+          pages: oldData.pages.map((page) => ({
+            ...page,
+            rows: page.rows.filter((member) => member.id !== state.deleteMember.user?.id),
+          })),
+        };
+      });
+
+      // Existing invalidations
+      Promise.all([utils.viewer.teams.get.invalidate(), utils.viewer.eventTypes.invalidate()]);
```

Comment:
Do we still need to invalidate these queries as well if the previous member is removed?

Reply:
I'd say so - yeah. Helps other pages be invalidated that use this value from cache